.github/workflows/security-trivy-scan-callable.yml

```diff
@@ -0,0 +1,123 @@
+name: Security - Scan Docker Image With Trivy
+
+on:
+  workflow_dispatch:
+    inputs:
+      image_ref:
+        description: 'Full image reference to scan e.g ghcr.io/n8n-io/n8n:latest'
+        required: true
+        default: 'ghcr.io/n8n-io/n8n:latest'
+  workflow_call:
+    inputs:
+      image_ref:
+        type: string
+        description: 'Full image reference to scan e.g ghcr.io/n8n-io/n8n:latest'
+        required: true
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
+
+jobs:
+  security_scan:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Pull Docker image
+        run: |
+          docker pull ${{ inputs.image_ref }}
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@6c175e9c4083a92bbca2f9724c8a5e33bc2d97a5 # v0.30.0
+        with:
+          image-ref: ${{ inputs.image_ref }}
+          format: 'json'
+          output: 'trivy-results.json'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW'
+          vuln-type: 'os,library'
+          ignore-unfixed: true
+          exit-code: '0'
+
+      - name: Process vulnerability results
+        id: process_vulns
+        run: |
+          if [ -f trivy-results.json ]; then
```

Comment:
I think best practice is to have a larger script in a standalone file that the GH action calls. More portable, easier to debug locally, syntax highlighting, etc.

Reply:
Do you mean just for the output processing? It would be a lot more work to do it locally because at this stage the image is already built. For local we would need to:
- Install Trivy 
- Build the image
- Scan the image